File: tipagens/05-Tipagens/05-Exerci╠ücios/01-Co╠üdigos incorretos/07-Ruby/exercise.rb
class Person
    attr_accessor :name, :age

    def initialize(name, age)
        @name = name
        @age = age
    end
end

def display_info(person)
    raise TypeError, "Expected Person instance" unless person.is_a?(Person)
    puts "Name: #{person.name}, Age: #{person.age}"
end

alice = Person.new("Alice", 30)
display_info(alice) # alice é uma classe/objeto Person

invalid_data = [1, 2, 3]
display_info(invalid_data) # invalid_data não é uma classe/objeto Person

result = "The number is: " + 10
puts result

File: tipagens/05-Tipagens/05-Exerci╠ücios/01-Co╠üdigos incorretos/02-Python/exercise.py
class Person:
    def __init__(self, name, age):
        self.name = name
        self.age = age

def display_info(person):
    if not isinstance(person, Person):
        raise TypeError("Expected a Person instance")
    print(f"Name: {person.name}, Age: {person.age}")

person1 = Person("Alice", 30)
display_info(person1)

result = "The age is: " + str(person1.age)
print(result)
